spec/models/spree/order_spec.rb

```diff
@@ -1211,76 +1211,73 @@
     end
   end
 
-  describe "a completed order with shipping and transaction fees" do
+  describe "#update_shipping_fees!" do
+    let(:distributor) { create(:distributor_enterprise) }
+    let(:order) {
+      create(:completed_order_with_fees, distributor:, shipping_fee:, payment_fee: 0)
+    }
+    let(:shipping_fee) { 5 }
+
+    it "does nothing if shipment is shipped" do
+      # An order need to be paid before we can ship a shipment
+      create(:payment, amount: order.total, order:, state: "completed")
+
+      shipment = order.shipments.first
+      shipment.ship
+
+      expect(shipment).not_to receive(:save)
+
+      order.update_shipping_fees!
+    end
+
+    it "saves the each shipment" do
+      order.shipments << create(:shipment, order:)
+      order.shipments.each do |shipment|
+        expect(shipment).to receive(:save)
+      end
+
+      order.update_shipping_fees!
+    end
+
+    context "with shipment including a shipping fee" do
+      it "updates shipping fee" do
+        # Manually udate line item quantity, in a normal scenario we would use
+        # order.contents method, which takes care of updating shipments
+        order.line_items.first.update(quantity: 2)
+
+        order.update_shipping_fees!
+
+        item_num = order.line_items.sum(&:quantity)
+        expect(order.reload.adjustment_total).to eq(item_num * shipping_fee)
```

Comment:
Because we're using test data, we should be able to specify the actual amounts here, which would be less error-prone than using a calculation. But let's not bother updating now..